lib/webview-helpers.js

```diff
@@ -16,23 +18,35 @@ let helpers = {};
 // same as chrome-backed webviews)
 // TODO: some of this function belongs in appium-adb
 async function webviewsFromProcs (adb, deviceSocket) {
+  const WEBVIEW_REGEXP = `@?webview_devtools_remote_(\\d+)`;
   let webviews = [];
   let out = await adb.shell(["cat", "/proc/net/unix"]);
   for (let line of out.split("\n")) {
     line = line.trim();
-    let webviewPid = line.match(/@?webview_devtools_remote_(\d+)/);
     if (deviceSocket) {
       if (line.indexOf(`@${deviceSocket}`) === line.length - deviceSocket.length - 1) {
-        if (webviewPid) {
-          webviews.push(WEBVIEW_BASE + webviewPid[1]);
-        } else {
+        let webviewPid;
+        let crosswalkWebviewSocket;
+
+        if (deviceSocket === "chrome_devtools_remote") {
           webviews.push(CHROMIUM_WIN);
+        } else if ((webviewPid = line.match(new RegExp(WEBVIEW_REGEXP)))) {
+          webviews.push(WEBVIEW_BASE + webviewPid[1]);
+        } else if ((crosswalkWebviewSocket = line.match(new RegExp("@" + CROSSWALK_REGEXP)))) {
+          webviews.push(WEBVIEW_BASE + crosswalkWebviewSocket[1]);
         }
       }
-    } else if (webviewPid) {
-      // for multiple webviews a list of 'WEBVIEW_<index>' will be returned
-      // where <index> is zero based (same is in selendroid)
-      webviews.push(WEBVIEW_BASE + webviewPid[1]);
+    } else {
+      let webviewPid;
+      let crosswalkWebviewSocket;
+
+      if ((webviewPid = line.match(new RegExp(WEBVIEW_REGEXP)))) {
```

Comment:
Sure - I originally pulled it up out of the conditional but didn't think it was as readable.

I'll have another go at refactoring.